File: thread/c/thread3.c
#include <stdio.h>
#include <unistd.h>
#include <pthread.h>


void* foo() {
    long tID = (long int) pthread_self();
    printf("foo => PID: %d, Thread ID: %ld\n", getpid(), tID);
    return NULL;
}

void* bar() {
    long tID = (long int) pthread_self();
    printf("bar => PID: %d, Thread ID: %ld\n", getpid(), tID);
    return NULL;
}

void* baz() {
    long tID = (long int) pthread_self();
    printf("baz => PID: %d, Thread ID: %ld\n", getpid(), tID);
    return NULL;
}

int main(void) {
    pthread_t t1, t2, t3;
    pthread_create(&t1, NULL, foo, NULL);
    pthread_create(&t2, NULL, bar, NULL);
    pthread_create(&t3, NULL, baz, NULL);
    pthread_join(t1, NULL);
    pthread_join(t2, NULL);
    pthread_join(t3, NULL);

    return 0;
}

/*
foo => PID: 6585, Thread ID: 6093860864
bar => PID: 6585, Thread ID: 6094434304
baz => PID: 6585, Thread ID: 6095007744
*/

File: thread/python/thread3.py
import threading


def foo():
    print("This is Foo.")
    print("Foo Thread ID:", threading.get_native_id())


def bar():
    print("This is Bar.")
    print("Bar Thread ID:", threading.get_native_id())


def baz():
    print("This is Baz.")
    print("Baz Thread ID:", threading.get_native_id())


if __name__ == "__main__":
    t1 = threading.Thread(target=foo).start()
    t2 = threading.Thread(target=bar).start()
    t3 = threading.Thread(target=baz).start()


"""
This is Foo.
Foo Thread ID: 927339
This is Bar.
Bar Thread ID: 927340
This is Baz.
Baz Thread ID: 927341
"""
